### PR TITLE
QDQ debugger - activations compare

### PR DESCRIPTION
**Pull request by chenfucn**
**Description**: Debugger for QDQ loss part 1

**Motivation and Context**
This is the first part of the QDQ debugger tool: activation comparison. The idea is that during quantization, we have an original float model and a qdq model. The debugger can run the two models side by side using the same input data. By comparing intermediate activations, we can help the model author figure out where the values differ, and take steps to reduce precision loss.

### CODE REVIEW

File: onnxruntime/python/tools/quantization/qdq_loss_debug.py [hunk 1]
```diff
@@ -145,3 +145,80 @@
...
+    pre_qdq_tensors: Optional[Sequence[numpy.ndarray]],
+    post_qdq_tensors: Optional[Sequence[numpy.ndarray]],
+) -> None:
+    if post_qdq_tensors and pre_qdq_tensors:
+        qdq_cmp[activation_name] = {}
+        qdq_cmp[activation_name]["pre_qdq"] = pre_qdq_tensors
+        qdq_cmp[activation_name]["post_qdq"] = post_qdq_tensors
+
+
+def activation_compare(
```
Comment (yufenglee) [inline onnxruntime/python/tools/quantization/qdq_loss_debug.py:165]: >activation_compare [](http://example.com/codeflow?start=4&length=18)

It doesn't compare value directly. How about giving it a better name, like: match_quant_tensor_with_tensor?

Comment (yufenglee) [reply onnxruntime/python/tools/quantization/qdq_loss_debug.py:165]: or create_activation_matching

**Comment by yufenglee** [comment]
LGTM. Please fix the Lint/Python format issue. Just FYI. It can usually pass the check after running black and isort:
https://github.com/microsoft/onnxruntime/blob/0c6037b5abe571fc43a55ef7a9d2f846820fbe5d/docs/Coding_Conventions_and_Standards.md#python-code-style

**Comment by chenfucn** [comment]
> LGTM. Please fix the Lint/Python format issue. Just FYI. It can usually pass the check after running black and isort: https://github.com/microsoft/onnxruntime/blob/0c6037b5abe571fc43a55ef7a9d2f846820fbe5d/docs/Coding_Conventions_and_Standards.md#python-code-style

I did run black and isort before each commit. still got those errors. I might have some configuration issues that I don't know about.

Many false positive in our pipeline linter though, it can't recognize onnx module, so any onnx reference would result in error.

**Pull request merged**